### PR TITLE
feat: report known malware for all ecosystems

### PR DESCRIPTION
**Pull request by behnazh-w**
If a package is already known to be malicious, this PR reports it as part of the `mcn_detect_malicious_metadata_1` check. Additionally, two new integration tests for known Python and npm malware have been added.

### CODE REVIEW

**Review by art1f1c3R — APPROVED** [review]

File: src/macaron/slsa_analyzer/checks/detect_malicious_metadata_check.py [hunk 1]
```diff
@@ -40,13 +42,16 @@ class MaliciousMetadataFacts(CheckFacts):
...
+
     #: Detailed information about the analysis.
     detail_information: Mapped[dict[str, JsonType]] = mapped_column(DBJsonDict, nullable=False)
 
-    #: The result of analysis, which is of dict[Heuristics, HeuristicResult] type.
-    result: Mapped[dict[Heuristics, HeuristicResult]] = mapped_column(
-        DBJsonDict, nullable=False, info={"justification": JustificationType.TEXT}
-    )
+    #: The result of analysis, which can be an empty dictionary.
+    result: Mapped[dict] = mapped_column(DBJsonDict, nullable=False, info={"justification": JustificationType.TEXT})
```
Comment (nicallen) [inline src/macaron/slsa_analyzer/checks/detect_malicious_metadata_check.py:54]: Why change `dict[Heuristics, HeuristicResult]]` to `dict`? In the new case it is empty, but that is still consistent with the old type.

File: src/macaron/util.py [hunk 2]
```diff
@@ -125,6 +125,80 @@ def send_get_http_raw(
     return response
 
 
+def send_post_http_raw(
```
Comment (nicallen) [inline src/macaron/util.py:128]: There is quite a bit of duplication between `send_post_http_raw`, `send_get_http_raw` and `send_get_http` (each has an implementation of error handling and retry logic), it would be good to refactor that at some point (but it doesn't have to be done as part of this change).

File: src/macaron/slsa_analyzer/checks/detect_malicious_metadata_check.py (hunk 1 above)
Comment (behnazh-w) [reply src/macaron/slsa_analyzer/checks/detect_malicious_metadata_check.py:54]: That's right. I don't need to change `dict[Heuristics, HeuristicResult]]`.

Comment (behnazh-w) [reply src/macaron/slsa_analyzer/checks/detect_malicious_metadata_check.py]: See commit [a61bb94](https://github.com/oracle/macaron/pull/922/commits/a61bb94fb8403268e0931445dc256b641f35e4d2).

**Review by nicallen — APPROVED** [review]

File: src/macaron/util.py [hunk 3]
```diff
@@ -125,6 +125,80 @@ def send_get_http_raw(
...
+    while response.status_code != 200:
+        logger.debug(
+            "Receiving error code %s from server.",
+            response.status_code,
+        )
+        if retry_counter <= 0:
+            logger.debug("Maximum retries reached: %s", error_retries)
+            return None
+        if response.status_code == 403:
+            check_rate_limit(response)
```
Comment (tromai) [inline src/macaron/util.py:187]: This could be a part of another PR. The logic for `check_rate_limit` was originally developed to handle limit exceeding in querying GithubAPI. Therefore going forward we need to decide / test if it's applicable to be used in this generic function call.

**Review by tromai — APPROVED** [review]
